src/sdk/Audio/BaseAudioPlayer.ts

```diff
@@ -23,11 +23,35 @@ export class BaseAudioPlayer {
     /**
      * Creates and initializes an instance of this class.
      * @constructor
+     * @param {AudioStreamFormat} audioFormat audio stream format recognized by the player.
      */
-    public constructor(audioFormat: AudioStreamFormat) {
+    public constructor(audioFormat?: AudioStreamFormat) {
+        if (audioFormat === undefined) {
+            audioFormat = AudioStreamFormat.getDefaultInputFormat();
+        }
         this.init(audioFormat);
     }
 
+    /**
+     * play Audio from result using safari-friendly audioContext methods
+     * @member BaseAudioPlayer.prototype.playResultAudio
+     * @function
+     * @public
+     * @param {ArrayBuffer} audioData audio data to be played.
+     */
+    public playResultAudio(audioData: ArrayBuffer): void {
```

Comment:
>playResultAudio [](start = 11, length = 15)

Why do we need a new public API for this?

Reply:
The clearBuffer() call in updateAudioBuffer on line 135 throws in Safari. I'm not sure how playAudioSample is currently being used in customer code, but will look and see if i can figure out the issue.

Reply:
Moved this to private method, public method will call this when webkit detected.